dipy/reconst/shm.py

```diff
@@ -1575,3 +1576,95 @@ def convert_sh_to_legacy(sh_coeffs, sh_basis, full_basis=False):
         raise ValueError("Invalid basis name.")
 
     return out_sh_coeffs
+
+
+def _setup_wigner_rotation(b, alpha, beta, gamma):
+    '''
+    Returns an array of all possible wigner-D matrices for a range of quantum numbers (l)
+
+    Parameters
+    ----------
+    b : int
+        The upper limit for the range of quantum numbers. The function generates Wigner-D
+        matrices for each quantum number from 0 to b-1.
+    alpha : float
+        The first Euler angle, representing a rotation around the z-axis.
+    beta : float
+        The second Euler angle, representing a rotation around the y-axis.
+    gamma : float
+        The third Euler angle, representing a rotation around the z-axis.
+
+    Returns
+    -------
+    Us : list of ndarray
```

Comment:
we avoid list of ndarray.  Can you just increase the dimension of your final array?

Reply:
For this array indexing is important, so a list of ndarray makes sense. Every index has a specific ndarray associated with it. Other alternatives could be to use hash maps.

Reply:
I understand. That's why I propose you to increase the dimension. 

from `[..., beta, alpha, gamma]`, create new array with `[..., beta, alpha, gamma, index]` where instead of list of ndarray which has memory issues.